doc.go

```diff
@@ -238,5 +238,31 @@ as well:
 	url, err := r.Get("article").URL("subdomain", "news",
 	                                 "category", "technology",
 	                                 "id", "42")
+
+Since **vX.Y.Z**, mux supports the addition of middlewares to a router, which are executed if a
+match is found (including subrouters). Middlewares are defined using the de facto standard function:
+
+   func(http.Handler) http.Handler
+
+Structs that implement the `Middleware` interface can also be used:
+
+	type Middleware interface {
+		Middleware(handler http.Handler) http.Handler
+	}
+
+Middlewares can be added to a router using `Router.AddMiddlewareFunc()` and
+`Router.AddMiddleware()`, respectively.
+
+A very basic middleware which logs the URI of the request being handled could be written as:
+
+	func simpleMw(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			log.Println(r.RequestURI)
+			next.ServeHTTP(w, r)
+		})
+	}
+
+Note: The handler chain will be stopped if your middleware doesn't call `next.ServeHTTP()`
```

Comment:
We should document/clarify how handlers are executed (in order they added).

Reply:
Roger, I'll clarify the calling order and stopping condition.